pkg/patterns/declarative/pkg/applier/direct.go

```diff
@@ -170,6 +170,7 @@ func (d *DirectApplier) Apply(ctx context.Context, opt ApplierOptions) error {
 		applyOpts.PruneResources = append(applyOpts.PruneResources, r...)
 	}
 
+	applyOpts.ServerSideApply = true
```

Comment:
Should this maybe be behind a feature flag, to allow users to choose when to enable this behavior? If so, could probably be done in the set of flags parsed some 20-ish lines above this one.

(I'm all for it, and would probably enable it first thing, but given that transitioning from CSA to SSA is not always straightforward, it might be worth giving the user a choice here; if nothing else, it enables bumping this library for other reasons without being forced to also manage the transition.)

Reply:
Thanks for the review @tomasaschan and sorry for the late reply. Changing this to a feature flag makes sense!  I changed it as a feature specific to the direct applier rather than another field in `ApplyOption` (interface level), because we cannot guarantee every implemented applier can support SSA. 

Also, since the SSA is a boolean type, to avoid confusing between unset vs default to false, I used a `preferred` option to distinguish whether it is configured or not.